### PR TITLE
Do not add index event listener if CCR disabled

### PR DESCRIPTION
**Pull request by Tim-Brooks**
Currently we add the `CcrRestoreSourceService` as a index event
listener. However, if ccr is disabled, this service is null and we
attempt to add a null listener throwing an exception. This commit only
adds the listener if ccr is enabled.

### CODE REVIEW

**Review by ywelsch — COMMENTED** [review]
Can you make sure we have a test where CCR is disabled?

**Comment by tbrooks8** [comment]
I've added a test @ywelsch

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xpack.core.XPackSettings;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+public class CcrDisabledIT extends ESIntegTestCase {
+
+    public void testClusterCanStartWithCcrInstalledButNotEnabled() throws Exception {
+        ensureGreen();
```
Comment (ywelsch) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java]: I think that the problem that is fixed here only occurs when an index is created, so let's create an index in this test. Perhaps also invoke the `XPackClient.prepareInfo()` API to check whether the plugin is disabled.

**Review by ywelsch — APPROVED** [review]

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java (hunk 1 above)
Comment (tbrooks8) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java]: I mean I specifically tried and the test failed without the my fix in the `Ccr.java` class. You can see it happens when the add listener method is called:

```
    public void addIndexEventListener(IndexEventListener listener) {
        ensureNotFrozen();
        if (listener == null) {
            throw new IllegalArgumentException("listener must not be null");
        }
        if (indexEventListeners.contains(listener)) {
            throw new IllegalArgumentException("listener already added");
        }

        this.indexEventListeners.add(listener);
    }
```

I will add the `prepareInfo` check.

Comment (tbrooks8) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java]: Also - it looks like CCR has not implemented the `XPackFeatureSet` functionality? When I search for references, it is not referenced in the ccr package.

Comment (jasontedor) [reply x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/CcrDisabledIT.java]: There is a PR open for this.

**Comment by tbrooks8** [comment]
gradle run build tests 1

**Comment by tbrooks8** [comment]
gradle run build tests 2

**Comment by tbrooks8** [comment]
run  gradle build tests 1

**Comment by tbrooks8** [comment]
run gradle build tests 2

**Comment by tbrooks8** [comment]
run gradle build tests 1

**Pull request merged**